lib/klaviyo/client.rb

```diff
@@ -1,74 +1,101 @@
-require 'open-uri'
 require 'base64'
+require 'cgi'
+require 'httparty'
 require 'json'
+require 'open-uri'
 
 module Klaviyo
   class KlaviyoError < StandardError; end
-  
+
   class Client
-    def initialize(api_key, url = 'http://a.klaviyo.com/')
-      @api_key = api_key
-      @url = url
+    include HTTParty
+
+    attr_reader :public_api_key, :private_api_key, :url
+
+    def initialize(public_api_key, private_api_key)
+      @public_api_key = public_api_key
+      @private_api_key = private_api_key
+      @url = 'https://a.klaviyo.com/api/'
     end
-    
+
     def track(event, kwargs = {})
       defaults = {:id => nil, :email => nil, :properties => {}, :customer_properties => {}, :time => nil}
       kwargs = defaults.merge(kwargs)
-      
+
       if kwargs[:email].to_s.empty? and kwargs[:id].to_s.empty?
         raise KlaviyoError.new('You must identify a user by email or ID')
       end
-      
+
       customer_properties = kwargs[:customer_properties]
       customer_properties[:email] = kwargs[:email] unless kwargs[:email].to_s.empty?
       customer_properties[:id] = kwargs[:id] unless kwargs[:id].to_s.empty?
 
       params = {
-        :token => @api_key,
+        :token => public_api_key,
         :event => event,
         :properties => kwargs[:properties],
-        :customer_properties => customer_properties,
-        :ip => ''
+        :customer_properties => customer_properties
       }
       params[:time] = kwargs[:time].to_time.to_i if kwargs[:time]
-     
+
       params = build_params(params)
-      request('api/track', params)
+
+      if kwargs['__track_once__']
+        request('track-once', params)
+      else
+        request('track', params)
+      end
     end
-    
+
     def track_once(event, opts = {})
       opts.update('__track_once__' => true)
       track(event, opts)
     end
-    
+
     def identify(kwargs = {})
       defaults = {:id => nil, :email => nil, :properties => {}}
       kwargs = defaults.merge(kwargs)
-      
+
       if kwargs[:email].to_s.empty? and kwargs[:id].to_s.empty?
         raise KlaviyoError.new('You must identify a user by email or ID')
       end
-      
+
       properties = kwargs[:properties]
       properties[:email] = kwargs[:email] unless kwargs[:email].to_s.empty?
       properties[:id] = kwargs[:id] unless kwargs[:id].to_s.empty?
 
       params = build_params({
-        :token => @api_key,
+        :token => public_api_key,
         :properties => properties
       })
-      request('api/identify', params)
+      request('identify', params)
+    end
+
+    def get_metrics
```

Comment:
a nit in styling on this could we create unique classes for each endpoint so something like:

`Klaviyo.Public.track/Identify`
`Klaviyo.Lists.list_functions_here`